drivers/sensors/l86xxx_uorb.c

```diff
@@ -0,0 +1,802 @@
+      return 0;
+    }
+
+  /* Setting baud rate also doesn't send an ACK but the interface baud rate
+  needs to be updated
+  */
+
+  if (cmd == SET_NMEA_BAUDRATE)
+  {
+    nxsig_usleep(20000); /* Should wait for a bit before changing interface baud rate */
+    ret = set_baud_rate(dev, (int)arg);
+    nxmutex_unlock(&dev->devlock);
+    return ret;
+  }
+
+  /* Some commands will send ACKs,
+  wait for them here before unlocking the mutex
+  */
+
+  /* ACK message will be $PMTK001,<cmd num>,<flag>
+  flag num indicates success of command
+  0 = Invalid packet
+  1 = Unsupported packet type
+  2 = Valid packet, but action failed
+  3 = Valid packet, action succeeded
+  */
+
+  memset(buf, '\0', 50);
+  snprintf(buf, 50, "$PMTK001,%d", cmd);
+  sninfo("Waiting for ACK from L86...\n");
+  for (; ; )
+    {
+      read_line(dev);
+      if (strncmp(buf, dev->buffer, strlen(buf)) == 0) break;
+    }
+
+  sninfo("ACK received!\n");
+  nxmutex_unlock(&dev->devlock);
+
+  /* Flag num is always in position 13 of ack, subtract by '0'
+  to obtain return val
+  */
+
+  ret = dev->buffer[13] - '0';
+  if (ret == 1)
+    {
+      return -ENOSYS;
+    }
+  else if (ret == 2)
+    {
+      return -EIO;
+    }
+  else if (ret == 3)
+    {
+      return 0;
+    }
+  else
+    {
+      return -EINVAL;
+    }
+}
+
+/****************************************************************************
+ * Name: read_line
+ *
+ * Description:
+ *   Reads line from UART interface and stores in priv buffer
+ *
+ * Arguments:
+ *    dev       -  Pointer L86-XXX priv struct
+ *
+ * Returns:
+ *  Negative number if reading from device failed, else number
+ *  of bytes read
+ ****************************************************************************/
+
+static int read_line(l86xxx_dev_s *dev)
+{
+  memset(dev->buffer, '\0', sizeof(dev->buffer));
+  int line_len = 0;
+  int err;
+  char next_char;
+  do
+    {
+      err = file_read(&dev->uart, &next_char, 1);
+      if (err < 0)
+        {
+          snerr("Couldn't read from UART device: %d\n", err);
+          return err;
+        }
+
+      if (next_char != '\r' && next_char != '\n')
+        {
+          dev->buffer[line_len++] = next_char;
+        }
+    }
+  while (next_char != '\r' && next_char != '\n'
+             && line_len < sizeof(dev->buffer));
+  dev->buffer[line_len] = '\0';
+  return line_len;
+}
+
+/****************************************************************************
+ * Name: l86xxx_control
+ *
+ * Description:
+ *   Send commands to the l86xxx GNSS module
+ *
+ * Returns:
+ *   Return value from send_command
+ ****************************************************************************/
+
+static int l86xxx_control(FAR struct sensor_lowerhalf_s *lower,
+                        FAR struct file *filep, int cmd, unsigned long arg)
+{
+  FAR l86xxx_dev_s *dev = container_of(lower, FAR l86xxx_dev_s, lower);
+  return send_command(dev, (L86XXX_PMTK_COMMAND)cmd, arg);
+}
```

Comment:
The command here actually needs to be an `SNIOC` command I believe; could you just change your PMTK enum to `SNIOC` `#defines`? I think it's in `include/nuttx/sensors.h`?

Reply:
I've added missing SNIOC command equivalents for the commands I've added, and changed the control function to use SNIOC